instaloader/structures.py

```diff
@@ -251,30 +250,31 @@ def typename(self) -> str:
         """Type of post, GraphImage, GraphVideo or GraphSidecar"""
         return self._field('__typename')
 
-    def get_sidecar_nodes(self) -> Iterator[PostSidecarNode]:
+    def get_sidecar_nodes(self, start=0, end=0) -> Iterator[PostSidecarNode]:
```

Comment:
Instead of getting all the nodes, I've added here `start` and `end` parameters, by default setted to 0. 
I've decided to do the job of selecting the posts here, because in this function I have access to the `edges`. Here, I know how many images are inside the post and therefore I can easily work with indexes.

Reply:
I like that the slicing happens here and it seems cleanly implemented. However the default should return all sidecar nodes, we cannot change the default behavior of a function.

I think it would be much more intuitive if the *start* and *end* parameters were indices with the same meaning as in Python array slicing, i.e. *0* is the first item, *1* is the second item, *-1* the last, and so on. This would not require any further explanation as this is what Python developers are used to. Also the implementation is simple: The if comparison below would then be a simple `start <= idx <= end`, and a simple `start %= len(edges)`, `end %= len(edges)` would suffice to handle negative indices. *start* would default to 0, *end* to -1. I think this will also simplify the big if in the Instaloader constructor.